onnxruntime/core/session/inference_session.h

```diff
@@ -780,7 +776,7 @@ class InferenceSession {
   // "session.use_ort_model_bytes_directly" to "1", this will be empty
   std::vector<uint8_t> ort_format_model_bytes_data_holder_;
 
-  std::shared_ptr<onnxruntime::AllocatorManager> allocator_manager_;
+  AllocatorManager allocator_manager_;
```

Comment:
An EP could live longer than a session. So, what if an EP is still alive but the allocator manager is gone?